src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -307,65 +321,79 @@ protected function guardAgainstInvalidTranslationResetRequest($content)
     /**
      * @param $lang
      * @param $type
-     * @param $selected
+     * @param string $theme Selected theme name
      * @param null $search
      *
      * @return array
      */
-    private function getNormalTree($lang, $type, $selected, $search = null)
+    private function getNormalTree($lang, $type, $theme, $search = null)
     {
-        $treeBuilder = new TreeBuilder($this->translationService->langToLocale($lang), $selected);
-        $catalogue = $this->translationService->getTranslationsCatalogue($lang, $type, $selected, $search);
+        $treeBuilder = new TreeBuilder($this->translationService->langToLocale($lang), $theme);
+        $catalogue = $this->translationService->getTranslationsCatalogue($lang, $type, $theme, $search);
 
-        return $this->getCleanTree($treeBuilder, $catalogue, $type, $selected, $search);
+        return $this->getCleanTree($treeBuilder, $catalogue, $theme, $search);
     }
 
     /**
-     * @param $lang
-     * @param $type
-     * @param $selected
-     * @param null $search
+     * @param string $lang Two-letter iso code
+     * @param string $selectedModuleName Selected module name
+     * @param string|null $search
      *
      * @return array
      */
-    private function getModulesTree($lang, $type, $selected, $search = null)
+    private function getModulesTree($lang, $selectedModuleName, $search = null)
     {
+        $theme = null;
         $locale = $this->translationService->langToLocale($lang);
+
         $moduleProvider = $this->container->get('prestashop.translation.external_module_provider');
-        $moduleProvider->setModuleName($selected);
+        $moduleProvider->setModuleName($selectedModuleName);
 
-        $treeBuilder = new TreeBuilder($locale, $this->getSelectedTheme());
+        $treeBuilder = new TreeBuilder($locale, $theme);
```

Comment:
So `$theme` is always `null` ?

Reply:
Yes when translating modules. Otherwise you'd have to translate your modules again for every theme (including if you switch themes). With this change only the wordings used in module templates overridden by a theme will be allowed to have a specific translation for that theme. This could be improved later by adding an option that brings ALL wordings, including core and modules, and that allows translating them for a specific theme.

Reply:
then you don't need this `$theme` variable you can simply call `$treeBuilder = new TreeBuilder($locale, null);`

Reply:
I did it so that the parameter role is evident, but it could be removed as well later.